### PR TITLE
OpenAI API client upgrade and code migration

### PR DESCRIPTION
**Pull request by bsbodden**
Upgrade OpenAI Python lib to >1.0.0 and migrates the code following https://github.com/openai/openai-python/discussions/742#async-client

Also, conditional skipping logic for vectorizers has been fixed (it was always skipping)

### CODE REVIEW

File: docs/user_guide/llmcache_03.ipynb [hunk 1]
```diff
@@ -35,11 +35,13 @@
     "\n",
     "api_key = os.getenv(\"OPENAI_API_KEY\") or getpass.getpass(\"Enter your OpenAI API key: \")\n",
     "\n",
-    "openai.api_key = api_key\n",
+    "client = OpenAI(\n",
+    "  api_key=os.environ['OPENAI_API_KEY'],  # this is also the default, it can be omitted\n",
```
Comment (tylerhutcherson) [inline docs/user_guide/llmcache_03.ipynb]: We need to pass the `api_key` object extracted from above into here

File: requirements-all.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-openai<=0.28.1
+openai<=1.13.3
```
Comment (tylerhutcherson) [inline requirements-all.txt]: Wondering if it makes sense to pin in the opposite direction? Maybe >= `1.13.0`

File: tests/integration/test_vectorizers.py [hunk 3]
```diff
@@ -9,17 +9,12 @@
...
 def skip_vectorizer() -> bool:
     # os.getenv returns a string
     v = os.getenv("SKIP_VECTORIZERS", "False").lower() == "true"
     print(v, flush=True)
-    return v
-
-
-skip_vectorizer_test = lambda: pytest.config.getfixturevalue("skip_vectorizer")
-
+    return True
```
Comment (tylerhutcherson) [inline tests/integration/test_vectorizers.py]: Wouldn't this always be true and thus never run the test?

**Review by tylerhutcherson — CHANGES_REQUESTED** [review]

File: requirements-all.txt (hunk 2 above)
Comment (bsbodden) [reply requirements-all.txt]: Indeed, I missed that

File: tests/integration/test_vectorizers.py (hunk 3 above)
Comment (bsbodden) [reply tests/integration/test_vectorizers.py]: Bob Saget! sorry, that was me debugging and leaving the test behind

**Review by tylerhutcherson — APPROVED** [review]

**Comment by tylerhutcherson** [comment]
Once CI passes, good to go. Will merge.

**Pull request merged**